transport/tchannel/outbound.go

```diff
@@ -133,10 +133,17 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		return nil, err
 	}
 
+	var headerItems map[string]string
+	if rawHeader {
+		headerItems = req.Headers.ForwardingItems()
+	} else {
+		headerItems = req.Headers.Items()
+	}
+
 	// Inject tracing system baggage
-	reqHeaders := tchannel.InjectOutboundSpan(call.Response(), req.Headers.Items())
+	reqHeaders := tchannel.InjectOutboundSpan(call.Response(), headerItems)
 
-	if err := writeRequestHeaders(ctx, format, reqHeaders, call.Arg2Writer); err != nil {
+	if err := writeRequestHeaders(ctx, format, reqHeaders, call.Arg2Writer, rawHeader); err != nil {
```

Comment:
Why do we thread `rawHeader` here? The decision was already made above when extracting the map[string]string from req.Headers.